helix-core/src/main/java/org/apache/helix/monitoring/metrics/model/Metric.java

```diff
@@ -31,11 +31,6 @@
    */
   String getMetricName();
 
-  /**
-   * Resets the internal state of this metric.
-   */
-  void reset();
```

Comment:
Should we keep this interface here to preserve the generality of the reset method? There could be other Metrics in addition to CountMetric and LatencyMetric that might benefit from reset() method.

But this is a might-be, so we could keep this and perhaps move it back to the interface when we actually need it.

Reply:
As we discussed offline, we will remove this method for now.
1. the reset is not used for now.
2. the current implementation of reset in the latency metric is not doing the expected resetting logic. So we should change it to a private method.